logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/Conversation.kt

```diff
@@ -3,15 +3,45 @@ package com.wire.kalium.logic.data.conversation
 import com.wire.kalium.logic.data.id.PlainId
 import com.wire.kalium.logic.data.id.QualifiedID
 import com.wire.kalium.logic.data.id.TeamId
+import com.wire.kalium.logic.data.publicuser.model.OtherUser
 import com.wire.kalium.logic.data.user.User
 import com.wire.kalium.logic.data.user.UserId
-
 typealias ConversationId = QualifiedID
 
 data class Conversation(val id: ConversationId, val name: String?, val type: Type, val teamId: TeamId?) {
     enum class Type { SELF, ONE_ON_ONE, GROUP }
 }
 
+sealed class ConversationDetails(val conversation: Conversation) {
+
+    class Self(conversation: Conversation) : ConversationDetails(conversation)
+
+    class OneOne(
+        conversation: Conversation,
+        val otherUser: OtherUser,
+        val connectionState: ConnectionState,
+        val federationStatus: FederationStatus,
+        val legalHoldStatus: LegalHoldStatus
```

Comment:
Does `legalHoldStatus` here refer to the `legalHoldStatus` of the other user of is it something else?

Reply:
Yep.

It's supposed to enable consumers to more easily display an indication on whether this conversation is being recorded or not, based on the members.